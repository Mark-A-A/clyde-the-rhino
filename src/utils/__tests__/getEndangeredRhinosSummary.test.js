const { getEndangeredRhinosSummary } = require('../getEndangeredRhinosSummary');

const rhinoceroses = require('../../data');

describe("getEndangeredRhinosSummary =>", () => {
  it("should return endangered rhinos summary", () => {

    const summary = getEndangeredRhinosSummary(rhinoceroses);

    console.dir(summary)
    expect(summary.white_rhinoceros).toBeUndefined();
    expect(summary.white_rhinoceros).toBeUndefined();
    expect(summary.javan_rhinoceross).toBeUndefined();
    expect(summary.indian_rhinoceros.rhinos).toEqual([
      {
        name: 'Phil',
        id: expect.anything(),
      }
    ]);
    expect(summary.indian_rhinoceros.count).toBe(1);
    expect(summary).toMatchSnapshot();
  });
});

