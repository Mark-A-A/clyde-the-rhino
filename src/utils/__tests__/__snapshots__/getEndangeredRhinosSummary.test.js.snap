// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEndangeredRhinosSummary => should return endangered rhinos summary 1`] = `
Object {
  "indian_rhinoceros": Object {
    "count": 1,
    "rhinos": Array [
      "Phil",
    ],
  },
  "javan_rhinoceros": undefined,
  "white_rhinoceros": undefined,
}
`;
